backend/sites/src/app/services/site/site.service.ts

```diff
@@ -1116,4 +1122,55 @@ export class SiteService {
       }
     }
   }
+
+  /**
+   * Processes and saves site disclosure based on the provided actions.
+   * @param siteDisclosure - Site disclosure data including actions to be performed.
+   * @param userInfo - Information about the user performing the actions.
+   * @param transactionalEntityManager - Entity manager for handling transactions.
+   */
+  async processSiteDisclosure(
```

Comment:
Does this method need unit tests?

Reply:
Yeah, I forget that I will create test case for this

Reply:
Done